include/caffe/filler.hpp

```diff
@@ -157,6 +157,27 @@ class XavierFiller : public Filler<Dtype> {
   }
 };
 
+/**
+ * @brief Fills a Blob with values @f$ x \sim N(-a, +a) @f$ where @f$ a @f$ is
+ *        set by the number of incoming nodes, based on the paper [He,
+ *        Zhang, Ren and Sun 2015]
+ */
+template <typename Dtype>
+class MSRFiller : public Filler<Dtype> {
+ public:
+  explicit MSRFiller(const FillerParameter& param)
+      : Filler<Dtype>(param) {}
+  virtual void Fill(Blob<Dtype>* blob) {
+    CHECK(blob->count());
+    int fan_in = blob->count() / blob->num();
```

Comment:
@nickcarlevaris I havn't read those. It might be better to name it like `MSRForwardFiller`? Anyways, this PR should be helpful since we no longer need to set the filler variances by hand. Thanks!